Modules/Core/Common/include/itkShapedImageNeighborhoodRange.h

```diff
@@ -388,9 +388,13 @@ class ShapedImageNeighborhoodRange final
      */
     QualifiedIterator() = default;
 
-    /** Constructor that allows implicit conversion from non-const to const
-     * iterator. Also serves as copy-constructor of a non-const iterator.  */
-    QualifiedIterator(const QualifiedIterator<false> & arg) noexcept
+    QualifiedIterator(const QualifiedIterator<VIsConst> & arg) noexcept = default;
+
+    /** Constructor that for implicit conversion from non-const to const
+     * iterator.  */
+    template <bool TIsConst>
+    QualifiedIterator(
+      typename std::enable_if_t<!TIsConst && !VIsConst, const QualifiedIterator<false> &>::value & arg) noexcept
```

Comment:
I think it's a good idea to use SFINAE in this case. 👍  It only needs to be adjusted a little bit, to make it work!